docs: add Autoscaler telemetry documentation to website. (#8733)
hashicorp/nomad: website/pages/docs/autoscaling/telemetry.mdx

---
layout: docs
page_title: Telemetry
sidebar_title: Telemetry
description: >
  Overview of runtime metrics available in the Nomad Autoscaler.
---

# Nomad Autoscaler Telemetry

The Nomad Autoscaler agent collects various runtime metrics about the performance
of different libraries and subsystems. These metrics are aggregated on a ten
second interval and are retained for one minute. To configure the telemetry output
please see the [agent configuration][agent_telemetry_config].

> This data can be accessed via an HTTP endpoint, via sending a signal to the
> Nomad Autoscaler process or via a number of integrations.

Reviewer: ```suggestion
This data can be accessed via the `/v1/metrics` HTTP endpoint, via sending a 
signal to the Nomad Autoscaler process or via a number of integrations.
```

      </td>
      <td>Number of nanoseconds to complete the last GC cycle</td>
      <td>Timer</td>
    </tr>
  </tbody>
</table>

## Policy Metrics

Policy metrics provide insights into the performance of the Nomad Autoscaler's
policy handling.

<table>
  <thead>
    <tr>
      <th>Metric</th>
      <th>Description</th>
      <th>Type</th>
      <th>Labels</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <code>nomad-autoscaler.policy.total_num</code>
      </td>
      <td>The number of policies currently held within the autoscaler</td>
      <td>Gauge</td>
      <td></td>
    </tr>
    <tr>
      <td>
        <code>nomad-autoscaler.policy.source.error_count</code>
      </td>
      <td>Tracks the number of errors generated by the policy sources</td>
      <td>Counter</td>
      <td>policy_source</td>
    </tr>
  </tbody>
</table>

## Scaling Metrics

Scaling metrics provide insight into the performance of scaling actions as well
as overall success and failure counters.

<table>
  <thead>
    <tr>
      <th>Metric</th>
      <th>Description</th>
      <th>Type</th>
      <th>Labels</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>

>         <code>nomad-autoscaler.scale.evaluate_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.scale.evaluate_ns</code>
```

      </td>
      <td>The time taken to evaluate the checks within a single policy</td>
      <td>Timer</td>
      <td>policy_id, target_name</td>
    </tr>
    <tr>
      <td>

>         <code>nomad-autoscaler.scale.invoke_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.scale.invoke_ns</code>
```

      </td>
      <td>The time taken to invoke scaling based on the scaling evaluations</td>
      <td>Timer</td>
      <td>policy_id, target_name</td>
    </tr>
    <tr>
      <td>
        <code>nomad-autoscaler.scale.invoke.success_count</code>
      </td>
      <td>Tracks the number of successful scaling actions triggered</td>
      <td>Counter</td>
      <td></td>
    </tr>
    <tr>
      <td>
        <code>nomad-autoscaler.scale.invoke.error_count</code>
      </td>
      <td>Tracks the number of unsuccessful scaling actions triggered</td>
      <td>Counter</td>
      <td></td>
    </tr>
  </tbody>
</table>

## Plugin Metrics

Plugin metrics provide insight into the performance of Nomad Autoscaler plugins
and help identify potential bottle necks or latency issues.

<table>
  <thead>
    <tr>
      <th>Metric</th>
      <th>Description</th>
      <th>Type</th>
      <th>Labels</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>

>         <code>nomad-autoscaler.plugin.manager.access_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.plugin.manager.access_ns</code>
```

      </td>
      <td>The time taken to dispense a plugin</td>
      <td>Timer</td>
      <td></td>
    </tr>
    <tr>
      <td>

>         <code>nomad-autoscaler.target.status.invoke_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.target.status.invoke_ns</code>
```

      </td>
      <td>The time taken to perform the target plugin status call</td>
      <td>Timer</td>
      <td>policy_id, plugin_name</td>
    </tr>
    <tr>
      <td>

>         <code>nomad-autoscaler.target.scale.invoke_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.target.scale.invoke_ns</code>
```

      </td>
      <td>The time taken to perform the target plugin scale call</td>
      <td>Timer</td>
      <td>policy_id, plugin_name</td>
    </tr>
    <tr>
      <td>

>         <code>nomad-autoscaler.apm.query.invoke_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.apm.query.invoke_ns</code>
```

      </td>

>       <td>The time taken to perform the apm plugin query call</td>

Reviewer: ```suggestion
      <td>The time taken to perform the APM plugin query call</td>
```

      <td>Timer</td>
      <td>policy_id, plugin_name</td>
    </tr>
    <tr>
      <td>

>         <code>nomad-autoscaler.strategy.run.invoke_ms</code>

Reviewer: ```suggestion
        <code>nomad-autoscaler.strategy.run.invoke_ns</code>
```
